sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/codegen/UnsafeRowWriter.java

```diff
@@ -93,6 +93,26 @@ public void setNullAt(int ordinal) {
     Platform.putLong(holder.buffer, getFieldOffset(ordinal), 0L);
   }
 
+  @Override
+  public void setNullByte(int ordinal) {
```

Comment:
which is the reason why these methods have been introduced?

Reply:
These methods are needed for writing `UnsafeArrayData`, we fill the slot with 0s if we set it to null. The slot size in `UnsafeArrayData` is dependent on the data type we are storing in it.

I wanted to avoid writing a lot of duplicate code in the `InterpretedUsafeProjection`, and this is why I added these methods the `UnsafeWriter` parent class, and this also why they are in the `UnsafeRowWriter`.

Reply:
I see, but I am not sure about having only some of these methods here. I mean, in `UnsafeArrayData` we have also `setNullDouble`, `setNullFloat`, etc. etc. It seems a bit weird to me that some of them are set at the parent level and some other no. It's not a big deal but I'd prefer for consistency to have all of them. What do you think?

Reply:
We could also name them differently e.g.: `setNull1Byte`, `setNull2Bytes`, `setNull4Bytes` & `setNull8Bytes`

Reply:
I am not sure it is a good idea, because then everyone while writing code should know exactly how many bytes each type is. I prefer the current approach. I would rather either reintroduce the `setNullAt` method with a match in the `UnsafeArrayData`'s implementation or let is as it is now.

Reply:
This is pretty low level stuff, so you should know how many bytes things contain at this point. I'd rather leave as it is. Doing a type match on such a hot code path doesn't seem like a good idea.